qcodes/tests/drivers/test_keysight_b220x.py

```diff
@@ -0,0 +1,216 @@
+import pytest
+from pyvisa.errors import VisaIOError
+from qcodes.instrument_drivers.Keysight.keysight_b220x import KeysightB220X
+
+@pytest.fixture
+def uut() -> KeysightB220X:
+    try:
+        uut = KeysightB220X('switch_matrix',
+                            address='GPIB::22::INSTR')
+    except (ValueError, VisaIOError):
+        # Either there is no VISA lib installed or there was no real instrument found at the
+        # specified address => use simulated instrument
+        import qcodes.instrument.sims as sims
+        path_to_yaml = sims.__file__.replace('__init__.py', 'keysight_b220x.yaml')
+
+        uut = KeysightB220X('switch_matrix',
+                            address='GPIB::1::INSTR',
+                            visalib=path_to_yaml + '@sim'
+                            )
+
+    uut.get_status()
+    uut.clear_status()
+    uut.reset()
+
+    yield uut
+
+    uut.close()
+
+
+def test_idn_command(uut):
+    assert "AGILENT" in uut.IDN()['vendor']
+    assert 0 == uut.get_status()
+
+
+def test_connect(uut):
+    uut.connect(2, 48)
+    assert 0 == uut.get_status()
```

Comment:
in case the real instrument is connected, does it make sense to also assert its state? For example, after connecting these ports to assert the set of connections?

Reply:
I would love to do that, however this will only work with the real instrument. When using the pyvisa-sim simulated instrument it is not „smart enough“ to behave correctly. 

Of course I could work around that with some „if simulated:“ etc, but I thought it’s not really worth the effort. After all I mostly want to validate that the command is recognized as „valid“. I then have to trust in the manufacturer to react properly to the command.

Reply:
ok. i see.